File: frontend/src/api/notifications.ts
import { http, type ApiResponse } from './http'

export interface TeachingNotification {
  id: number
  policy_change_id: number | null
  notification_type: string
  level: 'urgent' | 'important' | 'normal' | 'observe' | string
  title: string
  content: string
  course_ids: number[]
  chapter_ids: number[]
  source_url: string | null
  action_url: string | null
  is_read: boolean
  read_time: string | null
  created_time: string
  updated_time: string
}

export const notificationApi = {
  list(unreadOnly = false, limit = 50) {
    return http.get<ApiResponse<TeachingNotification[]>>('/notifications', {
      params: { unread_only: unreadOnly, limit },
    })
  },
  markRead(id: number) {
    return http.post<ApiResponse<TeachingNotification>>(`/notifications/${id}/read`)
  },
  markAllRead() {
    return http.post<ApiResponse<{ updated: number }>>('/notifications/read-all')
  },
}

File: backend/app/schemas/notifications.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TeachingNotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    policy_change_id: int | None
    notification_type: str
    level: str
    title: str
    content: str
    course_ids: list[int]
    chapter_ids: list[int]
    source_url: str | None
    action_url: str | None
    is_read: bool
    read_time: datetime | None
    created_time: datetime
    updated_time: datetime


class NotificationReadUpdate(BaseModel):
    is_read: bool = True
